Add Data Package (v2) blog post (#919)
frictionlessdata/frictionlessdata.io: site/blog/2024-06-26-datapackage-v2-release/README.md

---
title: Data Package version 2.0 is out!
date: 2024-06-26
tags: ["news"]
category: news
image: /img/blog/DP-release.png
description: We are very excited to announce the release of Data Package v2
author: Sara Petti
---

> We are very excited to announce the release of the version 2.0 of the Frictionless Standard. Thanks to the generous support of [NLnet](https://nlnet.nl/) from November last year we were able to [focus on reviewing the Frictionless Standard](https://frictionlessdata.io/blog/2023/11/15/frictionless-specs-update/#additional-deliverables) in order to include features that were often requested throughout the years and improve extensibility for domain-specific implementations. 

Reviewer: Not sure if now we need to mix this Fricitionless / Data Package Standard. Maybe we just need to mention that during the work, the new name was established to avoid confusion.

Author: Do you think the "We are very excited to announce the release of the version 2.0 of Data Package (previously known as Frictionless Specs)" would work better?

Reviewer: I agree, something like this

If you are curious about the conversations we had during the Standard review,they were all captured (and recorded) in [the blog summaries of the community calls](https://frictionlessdata.io/blog/). Alternatively you can also check out the [closed issues on GitHub](https://github.com/frictionlessdata/datapackage/milestone/6?closed=1). 

## So what is new in version 2?

During these months we have been working on the core specifications that compose the Standard, namely: [Data Package](https://datapackage.org/standard/data-package/) – a simple container format for describing a coherent collection of data in a single ‘package’, [Data Resource](https://datapackage.org/standard/data-resource/) to describe and package a single data resource, [Table Dialect](https://datapackage.org/standard/table-dialect/) to describe how tabular data is stored in a file, and [Table Schema](https://datapackage.org/standard/table-schema/) to declare a schema for tabular data.

During the update process we tried to be as little disruptive as possible, avoiding breaking changes when possible. 

> We put a lot of effort into removing ambiguity, cutting or clarifying under-defined features, and promoting some well-oiled [recipes](https://datapackage.org/recipes/caching-of-resources/) into the Standard itself. Among other things, we added versioning, to ensure consistency and reliability over time, and added support for categorical data.

Reviewer: An example of a pattern that has been promoted - https://specs.frictionlessdata.io/patterns/#missing-values-per-field

Author: Super, thanks!

If you are curious and would like to know more details about what changes with version 2, go and have a look at the [Changelog](https://datapackage.org/overview/changelog/) we published.

> To increase and facilitate adoption, we published a [metadata mapper written in Python](https://github.com/frictionlessdata/dplib-py). We have also worked on Data Package integrations for the most notable open data portals out there. We therefore proposed a Data Package serializer to [Invenio RDM](https://inveniordm.web.cern.ch/), created a pull request that exposes `datapackage.json` as a metadata export target in the [Open Science Framework](https://www.cos.io/) system, and built an extension that adds a `datapackage.json` endpoint to every dataset in [CKAN](https://github.com/frictionlessdata/ckanext-datapackage).

Reviewer: I would mention Invenio RDM / Zenodo together

---
title: Data Package version 2.0 is out!
date: 2024-06-26
tags: ["news"]
category: news

> image: /img/blog/DP-release.png

Reviewer 2: Is this image added? I don't see it as part of this PR.

Author: It's added somewhere else: https://github.com/frictionlessdata/frictionlessdata.io/commit/3c3d3e438c8dddf7f6f53bb2eb226def6390ad19

description: We are very excited to announce the release of Data Package v2
author: Sara Petti
---

> We are very excited to announce the release of the version 2.0 of Data Package (previously known as Frictionless Specs). Thanks to the generous support of [NLnet](https://nlnet.nl/) from November last year we were able to [focus on reviewing Data Package](https://frictionlessdata.io/blog/2023/11/15/frictionless-specs-update/#additional-deliverables) in order to include features that were often requested throughout the years and improve extensibility for domain-specific implementations. 

Reviewer: 2.0 of the Data Package standard

> We are very excited to announce the release of the version 2.0 of Data Package (previously known as Frictionless Specs). Thanks to the generous support of [NLnet](https://nlnet.nl/) from November last year we were able to [focus on reviewing Data Package](https://frictionlessdata.io/blog/2023/11/15/frictionless-specs-update/#additional-deliverables) in order to include features that were often requested throughout the years and improve extensibility for domain-specific implementations. 

Reviewer 2: ```suggestion
We are very excited to announce the release of version 2.0 of Data Package (previously known as Frictionless Specs). Thanks to the generous support of [NLnet](https://nlnet.nl/) in November last year, we were able to include features that were often requested by users throughout the years and improve extensibility for domain-specific implementations. 
```

I don't think the anchor https://frictionlessdata.io/blog/2023/11/15/frictionless-specs-update/#additional-deliverables should be referenced. If you want to reference the previous blog post, I would link "[November last year]"

> Data Package is a standard for data containerisation, which consists of a set of simple yet extensible specifications to describe datasets, data files and tabular data. It is a data definition language (DDL) and data API that enhances data FAIRness (findability, accessibility, interoperability, and reusability). Since the last releases of Data Package and all its components, the community had requested a few features throught the years that would improve the standard support for specific data types and simplify the extensions. Some of the Internet requirements also have evolved meanwhile. We therefore used [the issues that had accumulated in the GitHub repository](https://github.com/frictionlessdata/datapackage/issues) to build our Roadmap.

Reviewer 2: ```suggestion
Data Package is a standard for data containerisation, which consists of a set of simple yet extensible specifications to describe datasets, data files and tabular data. It is a data definition language (DDL) and data API that enhances data FAIRness (findability, accessibility, interoperability, and reusability). Since its initial release in 2007, the community has suggested many features that could improve or extend the standard for use cases that weren't initially envisioned. Those were sometimes adopted, but there wasn't a versioning or governance process in place to truly evolve the standard.

We started with the [issues that had accumulated in the GitHub repository](https://github.com/frictionlessdata/datapackage/issues) to build our roadmap for v2. Many of the requested features are now adopted, making Data Package the answer for even more use cases.
```

> In parallel we assembled an outstanding Data Package Working Group composed of experts from the community. We carefully selected a diverse group of people who could bring to the discussion table different use-cases, formats, and data types that we would need the Standard to support, and crafted together with them [a governance model](https://datapackage.org/overview/governance/) that is explicit, in order to create an environment that adequately supports new contributions and ensures project sustainability.

Reviewer 2: ```suggestion
In parallel we assembled an outstanding Data Package Working Group composed of experts from the community. We carefully selected a diverse group of people who brought different use-cases, formats, and data types that we want the Standard to support. Together with them, we crafted [a governance model](https://datapackage.org/overview/governance/) that is explicit, in order to create an environment that adequately supports new contributions and ensures project sustainability.
```

We would like to thank each one of them for their remarkable contribution and for the incredibly insightful conversations we had during these months. Thank you to my colleague Evgeny Karev, Peter Desmet from the [Research Institute for Nature and Forest (INBO)](https://www.vlaanderen.be/inbo/en-gb/homepage/), Phil Schumm from [CTDS - University of Chicago](https://ctds.uchicago.edu/), Kyle Husmann from the [PennState University](https://www.psu.edu/), Keith Hughitt  from the [National Institutes of Health](https://www.nih.gov/), Jakob Voß from the [Verbundzentrale des GBV (VZG)](https://www.gbv.de/), Ethan Welty from the [World Glacier Monitoring Service](https://wgms.ch/), Paul Walsh from [Link Digital](https://linkdigital.com.au/), Pieter Huybrechts from the [Research Institute for Nature and Forest (INBO)](https://www.vlaanderen.be/inbo/en-gb/homepage/), Martin Durant from [Anaconda, inc.](https://www.anaconda.com/), Adam Kariv from [The Public Knowledge Workshop](https://www.hasadna.org.il/), Johan Richer from [Multi](https://www.multi.coop/?locale=fr), and Stephen Diggs from the [University of California Digital Library](https://cdlib.org/). 

> If you are curious about the conversations we had during the Standard review,they were all captured (and recorded) in [the blog summaries of the community calls](https://frictionlessdata.io/blog/). Alternatively you can also check out the [closed issues on GitHub](https://github.com/frictionlessdata/datapackage/milestone/6?closed=1). 

Reviewer 2: ```suggestion
If you are curious about the conversations we had during the Standard review, they are all captured (and recorded) in [the blog summaries of the community calls](https://frictionlessdata.io/blog/). Alternatively you can also check out the [closed issues on GitHub](https://github.com/frictionlessdata/datapackage/milestone/6?closed=1). 
```

## So what is new in version 2?

> During these months we have been working on the core specifications that compose the Standard, namely: [Data Package](https://datapackage.org/standard/data-package/) – a simple container format for describing a coherent collection of data in a single ‘package’, [Data Resource](https://datapackage.org/standard/data-resource/) to describe and package a single data resource, [Table Dialect](https://datapackage.org/standard/table-dialect/) to describe how tabular data is stored in a file, and [Table Schema](https://datapackage.org/standard/table-schema/) to declare a schema for tabular data.

Reviewer 2: ```suggestion
During these months we have been working on the core specifications that compose the Standard, namely:

- [Data Package](https://datapackage.org/standard/data-package/), a simple container format for describing a coherent collection of data in a single ‘package’
- [Data Resource](https://datapackage.org/standard/data-resource/) to describe and package a single data resource
- [Table Dialect](https://datapackage.org/standard/table-dialect/) to describe how tabular data is stored in a file
- [Table Schema](https://datapackage.org/standard/table-schema/) to declare a schema for tabular data
```

During the update process we tried to be as little disruptive as possible, avoiding breaking changes when possible. 

> We put a lot of effort into removing ambiguity, cutting or clarifying under-defined features, and promoting some well-oiled [recipes](https://datapackage.org/recipes/caching-of-resources/) into the Standard itself. An example of a recipe (or pattern, as they were called in v1) that has been promoted to the Standard is the [Missing values per field

Reviewer: I mean we didn't promote the Caching of Resources

Author: You are right! I wanted to link to recipes in general, not to that particular one. I'll remove the link.

> one](https://specs.frictionlessdata.io/patterns/#missing-values-per-field).

Reviewer 2: ```suggestion
](https://specs.frictionlessdata.io/patterns/#missing-values-per-field). We also added a versioning mechanism, support for categorical data, and changes that make it easier to extend the Standard.
```

> Among other things, we added versioning, to ensure consistency and reliability over time, and support for categorical data.

Reviewer 2: ```suggestion
```

Reviewer 2: Included in previous paragraph. Probably need to delete a stray empty line 28 too.

> If you are curious and would like to know more details about what changes with version 2, go and have a look at the [Changelog](https://datapackage.org/overview/changelog/) we published.

Reviewer 2: ```suggestion
If you would like to know the details about what has changed, see the [Changelog](https://datapackage.org/overview/changelog/) we published.
```

> To increase and facilitate adoption, we published a [metadata mapper written in Python](https://github.com/frictionlessdata/dplib-py). We have also worked on Data Package integrations for the most notable open data portals out there. Many people from the community use Zenodo, so we definitely wanted to target that. As they are moving their system to [Invenio RDM](https://inveniordm.web.cern.ch/), we proposed a Data Package serializer to the project. We also created a pull request that exposes `datapackage.json` as a metadata export target in the [Open Science Framework](https://www.cos.io/) system, and built an extension that adds a `datapackage.json` endpoint to every dataset in [CKAN](https://github.com/frictionlessdata/ckanext-datapackage).

Reviewer 2: ```suggestion
To increase and facilitate adoption, we published a [metadata mapper written in Python](https://github.com/frictionlessdata/dplib-py). We have also worked on Data Package integrations for the most notable open data portals out there. Many people from the community use Zenodo, so we definitely wanted to target that. They have recently migrated their infrastructure to [Invenio RDM](https://inveniordm.web.cern.ch/) and we proposed a Data Package serializer for better integration with the Standard. We also created a pull request that exposes `datapackage.json` as a metadata export target in the [Open Science Framework](https://www.cos.io/) system, and built an extension that adds a `datapackage.json` endpoint to every dataset in [CKAN](https://github.com/frictionlessdata/ckanext-datapackage).
```

Reviewer 2: The original text implied the move hasn't happened yet. I'm not sure about the details of the integration with Data Package, so please correct if needed.

Note: this is a feature that people reading the post will be interested in, so might be good to indicate a timeline of when this becomes available. Or that this will be announced in a future blog post.

Author: Very good point, I added a sentence about an upcoming blog with more info on that.

If you want to know more about how to coordinate a standard update, we shared our main takeaways at FOSDEM 2024. The presentation was recorded, and you can watch it [here](https://fosdem.org/2024/schedule/event/fosdem-2024-3109-updating-open-data-standards/).

## And what happens now?

> While the work on Data Package 2.0 is done (for now!), we will keep working on the [Data Package website and documentation](https://datapackage.org/) together with the working group, to make it as clear and straightforward as possible for newcomers. In parallel, we will also start integrating the version 2 changes in the [software implementations](https://frictionlessdata.io/universe/). 

Reviewer 2: ```suggestion
While the work on Data Package 2.0 is done (for now!), we will keep working on the [Data Package website and documentation](https://datapackage.org/) together with the Working Group, to make it as clear and straightforward as possible for newcomers. In parallel, we will also start integrating the version 2 changes in the [software implementations](https://datapackage.org/overview/software/).
```

Might be more useful to link to https://datapackage.org/overview/software/ than https://frictionlessdata.io/universe/ ?
